spec/controllers/admin/membership_submissions_controller_spec.rb

```diff
@@ -3,13 +3,107 @@
 RSpec.describe Admin::MembershipSubmissionsController, type: :controller do
   describe "GET #index" do
     it "redirects to admin dashboard" do
-      allow(@controller).to receive(:require_admin!).and_return(true)
+      authorize_admin!
       get :index
       expect(response).to redirect_to(admin_path)
     end
   end
 
-  pending "GET #pending"
-  pending "POST #pending"
-  pending "POST #reject"
+  describe "GET #pending" do
+    describe "when pending submissions are zero" do
+      before do
+        authorize_admin!
+        get :pending
+      end
+
+      it "sets the flash to indicate there are no submissions" do
+        expect(flash[:info]).to match(/no pending submissions/i)
+      end
+
+      it "redirects to admin path" do
+        expect(response).to redirect_to(admin_path)
+      end
+    end
+
+    describe "when pending submissions are greater than zero" do
+      let!(:slack_membership) { create(:slack_membership_submission) }
+
+      before do
+        authorize_admin!
+        get :pending
+      end
+
+      it "does not set flash/info" do
+        expect(flash[:info]).to be_blank
+      end
+
+      it "returns success" do
+        expect(response).to have_http_status(:success)
+      end
+    end
+  end
+
+  describe "GET #approve" do
+    describe "when pending submissions are zero" do
+      let!(:slack_membership) { create(:approved_slack_membership_submission) }
+
+      before do
+        authorize_admin!
+        get :approve, {params: {id: slack_membership.id}}
+      end
+
+      it "sets flash/warning to indicate previous status" do
+        expect(flash[:warning]).to match(/John Doe was previously approved/i)
```

Comment:
Let's either drop "John Doe" or set the name fields on line 48 above.  Would hate for this test to fail simply because we changed the factory used to create it.

Reply:
we already have names in factory, would be removing it.